fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashViewModel.java

```diff
@@ -0,0 +1,241 @@
+import com.github.dedis.popstellar.repository.LAOState;
+import com.github.dedis.popstellar.repository.remote.GlobalNetworkManager;
+import com.github.dedis.popstellar.utility.error.ErrorUtils;
+import com.github.dedis.popstellar.utility.error.keys.KeyException;
+import com.github.dedis.popstellar.utility.security.KeyManager;
+import com.google.gson.Gson;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import dagger.hilt.android.lifecycle.HiltViewModel;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+
+@HiltViewModel
+public class DigitalCashViewModel extends AndroidViewModel {
+  public static final String TAG = DigitalCashViewModel.class.getSimpleName();
+  /*
+   * Dependencies for this class
+   */
+  private final LAORepository laoRepository;
+  private final GlobalNetworkManager networkManager;
+  private final Gson gson;
+  private final KeyManager keyManager;
+  private final CompositeDisposable disposables;
+
+  private static final String LAO_FAILURE_MESSAGE = "failed to retrieve lao";
+  private static final String PUBLISH_MESSAGE = "sending publish message";
+
+  @Inject Wallet wallet;
+
+  //TODO are we in a row call
+  private RollCall rollCall;
+
+  private static final String DIGITAL_CASH = "DIGITAL_CASH";
+
+  /*
+   * LiveData objects for capturing events
+   */
+  private final MutableLiveData<SingleEvent<Boolean>> mOpenHomeEvent = new MutableLiveData<>();
+  private final MutableLiveData<SingleEvent<Boolean>> mOpenHistoryEvent = new MutableLiveData<>();
+  private final MutableLiveData<SingleEvent<Boolean>> mOpenSendEvent = new MutableLiveData<>();
+  private final MutableLiveData<SingleEvent<Boolean>> mOpenReceiveEvent = new MutableLiveData<>();
+  private final MutableLiveData<SingleEvent<Boolean>> mOpenIssueEvent = new MutableLiveData<>();
+  private final MutableLiveData<SingleEvent<Boolean>> mOpenReceiptEvent = new MutableLiveData<>();
+  private final MutableLiveData<SingleEvent<Boolean>> mSendNewTransactionEvent = new MutableLiveData<>();
+
+  private final LiveData<List<Lao>> mLAOs;
+  private final MutableLiveData<String> mLaoId = new MutableLiveData<>();
+  private final MutableLiveData<String> mLaoName = new MutableLiveData<>();
+
+  @Inject
+  public DigitalCashViewModel(
+      @NonNull Application application,
+      LAORepository laoRepository,
+      GlobalNetworkManager networkManager,
+      Gson gson,
+      KeyManager keyManager) {
+    super(application);
+    this.laoRepository = laoRepository;
+    this.networkManager = networkManager;
+    this.gson = gson;
+    this.keyManager = keyManager;
+    disposables = new CompositeDisposable();
+
+    this.mLAOs =
+        LiveDataReactiveStreams.fromPublisher(
+            this.laoRepository.getAllLaos().toFlowable(BackpressureStrategy.BUFFER));
+  }
+
+  /*
+   * Getters for MutableLiveData instances declared above
```

Comment:
Maybe add comment for events? at least which fragment is calling them? It becomes a mess to read and understand when they accumulate :)